### PR TITLE
PE-3127: implement sorting with new data table

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/2cq84v0o1rl68
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/39s2rrg6q2hh0

### CODE REVIEW

File: lib/pages/drive_detail/components/drive_detail_data_list.dart [hunk 1]
```diff
@@ -90,19 +90,51 @@ Widget _buildDataListContent(
...
+        if (item.type == 'folder') {
+          folders.add(item);
+        } else {
+          files.add(item);
+        }
+      }
+
+      // Sort folders alphabetically
+      folders.sort((a, b) {
+        int result = compareAlphabeticallyAndNatural(a.name, b.name);
```
Comment (karlprieb) [inline lib/pages/drive_detail/components/drive_detail_data_list.dart:107]: sort by date

File: lib/pages/drive_detail/components/drive_detail_data_list.dart [hunk 2]
```diff
@@ -90,19 +90,51 @@ Widget _buildDataListContent(
     leading: (file) => DriveExplorerItemTileLeading(
       item: file,
     ),
-    sort: (columnIndex) {
-      int sort(ArDriveDataTableItem a, ArDriveDataTableItem b) {
-        // TODO(add folders in the start of the list)
+    sortRows: (list, columnIndex, sort) {
```
[the remark this answers is not in the document]
Comment (thiagocarvalhodev) [reply lib/pages/drive_detail/components/drive_detail_data_list.dart]: bdd25db7

File: lib/pages/drive_detail/components/drive_detail_data_list.dart (hunk 1 above)
Comment (thiagocarvalhodev) [reply lib/pages/drive_detail/components/drive_detail_data_list.dart]: 4ac478cd

File: lib/pages/drive_detail/components/drive_detail_data_list.dart [hunk 3]
```diff
@@ -90,19 +90,23 @@ Widget _buildDataListContent(
...
-        if (columnIndex == 0) {
-          return compareAlphabeticallyAndNatural(a.name, b.name);
-        } else if (columnIndex == 1) {
-          return a.size.compareTo(b.size);
+    sortRows: (list, columnIndex, sortOrder) {
+      // Separate folders and files
+      List<ArDriveDataTableItem> folders = [];
+      List<ArDriveDataTableItem> files = [];
+
+      for (int i = 0; i < list.length; i++) {
```
Comment (karlprieb) [inline lib/pages/drive_detail/components/drive_detail_data_list.dart:98]: cache `list.length` for better perfomance

File: lib/pages/drive_detail/components/drive_detail_data_list.dart [hunk 4]
```diff
@@ -116,3 +120,40 @@ Widget _buildDataListContent(
...
   );
 }
+
+void _sortFoldersAndFiles(List<ArDriveDataTableItem> folders,
+    List<ArDriveDataTableItem> files, int columnIndex, TableSort sortOrder) {
+  _sortItems(folders, columnIndex, sortOrder);
+  _sortItems(files, columnIndex, sortOrder);
+}
+
+int _getResult(int result, TableSort sortOrder) {
```
Comment (karlprieb) [inline lib/pages/drive_detail/components/drive_detail_data_list.dart:130]: rename to `_ascDescSort`

Comment (thiagocarvalhodev) [reply lib/pages/drive_detail/components/drive_detail_data_list.dart]: [522b3f2](https://github.com/ardriveapp/ardrive-web/pull/929/commits/522b3f205f17403b1c7c2bdf14c08d93798ab437)

File: lib/pages/drive_detail/components/drive_detail_data_list.dart (hunk 3 above)
Comment (thiagocarvalhodev) [reply lib/pages/drive_detail/components/drive_detail_data_list.dart]: [27db196](https://github.com/ardriveapp/ardrive-web/pull/929/commits/27db196befdae7e1a6beceb51392e18375a4feb5)

**Review by karlprieb — APPROVED** [review]

**Pull request merged**